docs: created how to guide for deferred expressions (#4914)
ibis-project/ibis: docs/how_to/chain-expressions.md

# How to Chain Ibis Expressions using the Deferred Expression API (Also known as the Underscore (`_`) API)

Expressions can easily be chained using the deferred expression API, also known as the Underscore (`_`) API.

In this guide, we will look at ways that the `_` API can be used to conveniently create column expressions and then chain Ibis expressions.

## Setup

To get started, we need to import `_` from ibis:

```python
import ibis
from ibis import _

import pandas as pd
```

> Let's set up two tables using [ibis.MemTables](memtable-join.md) that were introduced in 3.2:

Reviewer: ```suggestion
Let's create two in-memory tables using [ibis.MemTables](memtable-join.md), an API introduced in 3.2:
```

> Let's set up two tables using [ibis.MemTables](memtable-join.md) that were introduced in 3.2:

Author: Fixed; thanks @cpcloud

```python
t1 = ibis.memtable(pd.DataFrame({'x': range(5), 'y': list('ab')*2 + list('e')}))
t2 = ibis.memtable(pd.DataFrame({'x': range(10), 'z': list(reversed(list('ab')*2 + list('e')))*2}))
```

## Creating ColumnExpressions

> We can use `_` to create ColumnExpressions without explicitly referencing TableExpression objects:

Reviewer: ```suggestion
We can use `_` to create new column expressions without explicit reference to the previous table expression:
```

```python
# We can pass a deferred expression into a function:

> def modf(t): return t.x % 3

Reviewer: ```suggestion
def modf(t):
    return t.x % 3
    
```

> # Chaining Expressions with Underscore

Reviewer: I ended up changing this to `Underscore` for readability.

> # How to Chain Ibis Expressions using the Deferred Expression API (Also known as the Underscore (`_`) API)

Reviewer: Let's make the title as concise as possible. How about this?

```suggestion
# Chaining Expressions with `_`
```

Author: I think I wanted to avoid just using `_` in the title for SEO purposes per #4704: people probably won't know to call it _ when wanting to chain expressions and I don't know how the _ character would affect searches, but that is basically pure speculation on my part.

What do you think? I do like the more concise title, mostly because the sidebar looks really cluttered right now, but also because I'm not a fan of really long titles.  I'm just worried about how people will _find_ this info.

Author: I guess the key is "chain Ibis expressions", and the "with _" will tell them "you can do it using this thing".  I also call out the actual name of the object in the text, so I'm going to go ahead and accept this one.

Thanks @cpcloud !

Reviewer: Yeah, it could also be `Chaining Expressions with Underscore`.
